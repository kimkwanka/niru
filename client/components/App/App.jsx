import React from 'react';
import Helmet from 'react-helmet';
import { renderRoutes } from 'react-router-config';
import { hot } from 'react-hot-loader/root';

import routes from 'Client/routes';

import Header from './Header/Header';
import Footer from './Footer/Footer';

const App = () => (
  <div className="App">
    <Helmet
      htmlAttributes={{ lang: 'en' }}
      titleTemplate="niru | %s"
      titleAttributes={{ itemprop: 'name', lang: 'en' }}
      meta={[
        { name: 'description', content: 'Universal fullstack boilerplate' },
      ]}
    />
    <Header />
    <main className="page-content relative overflow-hidden">
      {renderRoutes(routes)}
    </main>
    <Footer />
  </div>
);

export default hot(App);
