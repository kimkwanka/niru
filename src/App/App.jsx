import React from 'react';
import Helmet from 'react-helmet';
import Home from '../Home/Home';

const App = () => {

  return (
      <div className="App">
        <Helmet
          htmlAttributes={{ lang: 'en' }}
          titleTemplate="reniru | %s"
          titleAttributes={{ itemprop: 'name', lang: 'en' }}
          meta={[
            { name: 'description', content: 'Universal fullstack boilerplate' },
          ]}
        />
        <main>
          <Home />
        </main>
      </div>
  );
};

export default App;
