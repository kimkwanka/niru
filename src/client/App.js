/* eslint-disable react/jsx-filename-extension */
/* eslint-disable import/no-webpack-loader-syntax*/
/* eslint-disable react/prefer-stateless-function */

import React from 'react';
import { Provider } from 'react-redux';
import { Router, browserHistory } from 'react-router';
import getRoutes from '../shared/routes';
import { getHydratedStore } from '../shared/store';

//eslint-disable-next-line

const store = getHydratedStore();

export default class extends React.Component {
  render() {
    return (<Provider store={store}>
      <Router routes={getRoutes(store)} history={browserHistory} />
    </Provider>);
  }
}
