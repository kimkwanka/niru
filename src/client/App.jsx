import React from 'react';
import { Switch, Route } from 'react-router-dom';
import Helmet from 'react-helmet';
import PropTypes from 'prop-types';

import ScrollToTop from './components/shared/ScrollToTop';
import Header from './components/shared/Header';
import Footer from './components/shared/Footer';

import routesArr from '../shared/routes';

const App = ({ serverRoute }) => {
// If <App /> is rendered on the server we need to provide the serverRoute prop
// since StaticRouter can only render a single Route (Switch only works on client side).
// On the client though, just return all routes and let Switch do the work.

  const routes = serverRoute ?
    <Route key={serverRoute.path} {...serverRoute} /> :
    routesArr.map(route => <Route key={route.path} {...route} />);

  return (
    <ScrollToTop>
      <div className="App">
        <Helmet
          htmlAttributes={{ lang: 'en' }}
          titleTemplate="niru | %s"
          titleAttributes={{ itemprop: 'name', lang: 'en' }}
          meta={[
            { name: 'description', content: 'Universal fullstack boilerplate' },
          ]}
        />
        <Header />
        <main className="page-content relative overflow-hidden">
          <Switch>
            {routes}
          </Switch>
        </main>
        <Footer />
      </div>
    </ScrollToTop>
  );
};

App.propTypes = {
  serverRoute: PropTypes.objectOf(PropTypes.shape),
};

App.defaultProps = {
  serverRoute: null,
};

export default App;
