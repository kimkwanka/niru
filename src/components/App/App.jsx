import React from 'react';
import Helmet from 'react-helmet';
import { HashRouter as Router, Route, Link } from 'react-router-dom';

import Home from '../Home/Home';
import About from '../About/About';
import Styleguide from '../Styleguide/Styleguide';

const App = () => (
  <Router>
    <div className="App">
      <Helmet
        htmlAttributes={{ lang: 'en' }}
        titleTemplate="reniru | %s"
        titleAttributes={{ itemprop: 'name', lang: 'en' }}
        meta={[
          { name: 'description', content: 'Universal fullstack boilerplate' },
        ]}
      />
      <nav>
        <ul>
          <li>
            <Link to="/">Home</Link>
          </li>
          <li>
            <Link to="/about">About</Link>
          </li>
          <li>
            <Link to="/styleguide">Styleguide</Link>
          </li>
        </ul>
      </nav>
      <main>
        <Route path="/" exact component={Home} />
        <Route path="/about" component={About} />
        <Route path="/styleguide" component={Styleguide} />
      </main>
    </div>
  </Router>
);

export default App;
