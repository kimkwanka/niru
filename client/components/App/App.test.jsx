import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { App } from './App';

describe('<App /> - Snapshot', () => {
  it('matches its snapshot', () => {
    const wrapper = shallow(<App />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
