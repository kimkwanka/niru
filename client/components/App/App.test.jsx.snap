// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> - Snapshot matches its snapshot 1`] = `
<div
  className="App"
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
    htmlAttributes={
      Object {
        "lang": "en",
      }
    }
    meta={
      Array [
        Object {
          "content": "Universal fullstack boilerplate",
          "name": "description",
        },
      ]
    }
    titleAttributes={
      Object {
        "itemprop": "name",
        "lang": "en",
      }
    }
    titleTemplate="niru | %s"
  />
  <Header />
  <main
    className="page-content relative overflow-hidden"
  >
    <Switch>
      <Route
        exact={true}
        key="0"
        path="/"
        render={[Function]}
      />
      <Route
        key="1"
        path="/about"
        render={[Function]}
      />
      <Route
        key="2"
        path="/styleguide"
        render={[Function]}
      />
      <Route
        key="3"
        path="*"
        render={[Function]}
      />
    </Switch>
  </main>
  <Footer />
</div>
`;
